Library/Homebrew/missing_formula.rb

```diff
@@ -125,11 +125,23 @@ def deleted_reason(name)
         relative_path = path.relative_path_from tap.path
 
         tap.path.cd do
-          # We know this may return incomplete results for shallow clones but
-          # we don't want to nag everyone with a shallow clone to unshallow it.
-          log_command = "git log --name-only --max-count=1 --format=%H\\\\n%h\\\\n%B -- #{relative_path}"
-          hash, short_hash, *commit_message, relative_path =
-            Utils.popen_read(log_command).gsub("\\n", "\n").lines.map(&:chomp)
+          begin
+            timer_pid = fork do
+              # Let the user know what's going on when the search goes on for
+              # more than two seconds.
+              sleep 2
+              opoo "Searching through git history. This may take a while..."
```

Comment:
That seems a bit more accurate, yet at the same time it is somewhat strange. This is given, at least for me, that what triggers this is usually a mistaken key stroke. From that perspective it may be better to just leave it completely generic?...